cmake/hsBuildInfo.inl.cmake

```diff
@@ -0,0 +1,20 @@
+// Automatically generated for diff:
+// @CURRENT_DIFF_HASH@
+
+namespace plProduct
+{
+    constexpr uint32_t PRODUCT_BRANCH_ID = @PRODUCT_BRANCH_ID@;
+    constexpr uint32_t PRODUCT_BUILD_ID = @PRODUCT_BUILD_ID@;
+    constexpr uint32_t PRODUCT_BUILD_TYPE = @PRODUCT_BUILD_TYPE@;
+
+    static ST::string PRODUCT_CORE_NAME = ST_LITERAL("@PRODUCT_CORE_NAME@");
+    static ST::string PRODUCT_SHORT_NAME = ST_LITERAL("@PRODUCT_SHORT_NAME@");
+    static ST::string PRODUCT_LONG_NAME = ST_LITERAL("@PRODUCT_LONG_NAME@");
```

Comment:
Even though we can't have `constexpr` yet (😞), the strings should still be `const`